src/cs50/sql.py

```diff
@@ -280,29 +295,17 @@ def execute(self, sql, *args, **kwargs):
             # Infer whether app is defined
             assert flask.current_app
 
-            # If no connection for app's current request yet
-            if not hasattr(flask.g, "_connection"):
+            # Disconnect later - but only once
+            if not hasattr(self, "teardown_appcontext_added"):
```

Comment:
Should probably be `_teardown_appcontext_added`, no?

Reply:
Sure, changed.